### PR TITLE
sort

### PR DESCRIPTION
**Pull request by Boy-timi**

### CODE REVIEW

**Comment by Boy-timi** [comment]
let's me see!

File: codes/python/chapter_sorting/bubble_sort.py [hunk 1]
```diff
@@ -1,10 +1,47 @@
...
 '''
 
-import sys, os.path as osp
-sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))
 from include import *
+import sys
+import os.path as osp
+sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))
+
+"""冒泡排序"""
```
Comment (krahets) [inline codes/python/chapter_sorting/bubble_sort.py:12]: Two spaces are required, ie `""" 冒泡排序 """`

File: codes/python/chapter_sorting/quick_sort.py [hunk 2]
```diff
@@ -1,10 +1,132 @@
...
-import sys, os.path as osp
-sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))
 from include import *
+import sys
+import os.path as osp
+sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))
+
+
+"""快速排序类"""
+class quick_sort(object):
```
Comment (krahets) [inline codes/python/chapter_sorting/quick_sort.py:14]: class name should be like `class QuickSort:`

File: codes/python/chapter_sorting/quick_sort.py [hunk 3]
```diff
@@ -1,10 +1,132 @@
...
+        if (nums[left] > nums[mid]) ^ (nums[left] > nums[right]):
+            return left
+        elif (nums[mid] < nums[left]) ^ (nums[mid] > nums[right]):
+            return mid
+        return right
+
+    """哨兵划分（三数取中值）"""
+    def partition(self, nums, left, right):
+        # 以 nums[left] 作为基准数
+        med = self.median_three(nums, left, (left+right)//2, right)
```
Comment (krahets) [inline codes/python/chapter_sorting/quick_sort.py:59]: `(left + right) // 2`

File: codes/python/chapter_sorting/merge_sort.py [hunk 4]
```diff
@@ -1,10 +1,60 @@
...
+"""
+合并左子数组和右子数组
+左子数组区间 [left, mid]
+右子数组区间 [mid + 1, right]
+"""
+def merge(nums, left, mid, right):
+    # 初始化辅助数组 借助 copy模块
+    tmp = copy.deepcopy(nums[left:right+1])
+    # 左子数组的起始索引和结束索引
+    leftStart, leftEnd = left-left, mid - left
```
Comment (krahets) [inline codes/python/chapter_sorting/merge_sort.py:23]: All the **functions, variables** should following the format `left_start`

**Review by krahets — CHANGES_REQUESTED** [review]
Thanks for the hard work. There are still some comments need to be addressed. Thanks

File: codes/python/chapter_sorting/merge_sort.py [hunk 5]
```diff
@@ -1,10 +1,60 @@
...
+
+
+"""
+合并左子数组和右子数组
+左子数组区间 [left, mid]
+右子数组区间 [mid + 1, right]
+"""
+def merge(nums, left, mid, right):
+    # 初始化辅助数组 借助 copy模块
+    tmp = copy.deepcopy(nums[left:right+1])
```
Comment (krahets) [inline codes/python/chapter_sorting/merge_sort.py:21]: `tmp = nums[left:right+1]`

**Review by krahets — CHANGES_REQUESTED** [review]

**Pull request closed without merging**